src/RobotInterface/src/Helper.cpp

```diff
@@ -489,6 +504,21 @@ bool RobotInterface::setPositionReferences(const iDynTree::VectorDynSize& desire
             return false;
         }
         m_controlMode = VOCAB_CM_POSITION;
+
+        std::vector<yarp::dev::InteractionModeEnum>  modes;
+        std::vector<int>  jointsIndex;
+        for (int i=0;i<m_actuatedDOFs;i++) {
+            jointsIndex.at(i)=i;
+            if(m_isJointModeStiffVector.at(i)==true){
+            modes.at(i)=yarp::dev::InteractionModeEnum::VOCAB_IM_STIFF;
+            }
+            else {
+            modes.at(i)=yarp::dev::InteractionModeEnum::VOCAB_IM_COMPLIANT;
+            }
+        }
```

Comment:
I would write a function to set the interaction mode from the `walking-module`.
The function can be called `setInteractionMode()`. And the function should be called after the preparation of the robot.